parquet/src/main/java/org/apache/iceberg/parquet/MessageTypeToType.java

```diff
@@ -96,11 +96,13 @@ public Type struct(GroupType struct, List<Type> fieldTypes) {
 
   @Override
   public Type list(GroupType array, Type elementType) {
-    GroupType repeated = array.getType(0).asGroupType();
-    org.apache.parquet.schema.Type element = repeated.getType(0);
+    org.apache.parquet.schema.Type repeated = array.getType(0);
+    org.apache.parquet.schema.Type repeatedElement = array.getFields().get(0);
```

Comment:
From Parquet:

```java
  public Type getType(int index) {
    return fields.get(index);
  }

  public List<Type> getFields() {
    return fields;
  }
```

So `g.getType(0)` and `g.getFields().get(0)` will return the same object. That makes me suspect what's happening here. What was your intent with both `repeated` and `repeatedElement`?